x-pack/plugins/ml/public/explorer/explorer_utils.js

```diff
@@ -528,3 +565,38 @@ export async function loadTopInfluencers(selectedJobIds, earliestMs, latestMs, i
     }
   });
 }
+
+// Returns an array of influencer field names as strings for all selectedJobs
+// GET _ml/anomaly_detectors/<job-id>,<job-id>
+//   response format:
+// { count: 1, jobs: [ { "analysis_config": { influencers: [ 'influencerOne', 'influencerTwo' ] } } ] }
+export async function loadInfluencerFields(
+  selectedJobIds,
+  noInfluencersConfigured,
+) {
+  return new Promise((resolve) => {
+    const influencers = [];
+    if (noInfluencersConfigured !== true) {
+      ml.getMultipleJobs(
```

Comment:
Really great thoughts here, @jgowdyelastic. I think removing `getMultipleJobs` is the right call here since I can essentially access the already loaded jobs in this file via the `mlJobService` as you mentioned.

I've gone ahead and created a `getInfluencers` function in this file (`explorer_utils.js`) that is similar to `
selectedJobsHaveInfluencers` but returns an array of influencer name strings. This way I can use it similarly to how you mentioned - by grabbing all the influencers I need in `getIndexPattern` and also using `noInfluencersConfigured: (getInfluencers(selectedJobs).length === 0)` in lieu of `
selectedJobsHaveInfluencers`

Since that's the only context `selectedJobsHaveInfluencers` is being used (and only in `explorer.js`) I've removed that function since it's no longer needed. Let me know what you think.

Reply:
Updated in: a861b7d09a146889329f732a1f2fd651561a5086